docs/user-guide/multiple_sources.md

```diff
@@ -0,0 +1,89 @@
+# Multiple Sources for an Application
+
+Argo CD has the ability to specify multiple sources to add services to the Application. Argo CD compiles all the sources and reconciles each source individually for creating the application.
+
+You can provide multiple sources using the `sources` field. When you specify the sources field, Argo CD will ignore the values under `source` field for generating the application.
+
+See below example for specifying multiple sources:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: guestbook
+  namespace: argocd
+  labels:
+    argocd.argoproj.io/refresh: hard
+spec:
+  project: default
+  syncPolicy:
+    automated:
+      prune: true
+      selfHeal: true
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: argocd
+  sources:
+    - chart: elasticsearch
+      helm:
+        valueFiles:
+          - values.yaml
+      repoURL: https://helm.elastic.co
+      targetRevision: 7.6.0
+    - repoURL: https://github.com/argoproj/argocd-example-apps.git
+      path: guestbook
+      targetRevision: HEAD
+```
+
+The above example has 2 sources specified. Argo CD will reconcile each source separately and combine the resources that are generated for generating the application.
+
+In case application has multiple entries for the same source (repoURL), Argo CD would pick the source that is mentioned later in the list of sources. For example, consider the below list of sources:
+
+```yaml
+sources:
+    - chart: elasticsearch
+      helm:
+        valueFiles:
+          - values.yaml
+      repoURL: https://helm.elastic.co
+      targetRevision: 7.6.0
+    - repoURL: https://github.com/argoproj/argocd-example-apps.git
+      path: guestbook
+      targetRevision: HEAD
+    - chart: elasticsearch
+      helm:
+        valueFiles:
+          - values.yaml
+      repoURL: https://helm.elastic.co
+      targetRevision: 7.7.0
+```
+
+In the above list, the application has 2 sources referring to the same repoURL. In this case, Argo CD will generate the manifests for source with `targetRevision: 7.6.0` and then append the manifests generated for source with `targetRevision: 7.7.0`. 
+
+
+## Helm Value files from external Git repository
+
+Users can now provide provide value files to the helm repositories from external sources. See below example ApplicationSpec for the same,
+
+```yaml
+spec:
+  project: default
+  sources:
+  - repoURL: 'https://prometheus-community.github.io/helm-charts'
+    chart: prometheus
+    targetRevision: 15.6.0
+    ref: prometheus
```

Comment:
This example should be updated, since we're not planning to support Helm charts as `ref` sources.

Reply:
@crenshaw-dev "not planning" as in "not now" or as in "won't be considered to be implemented at all"?

I was looking forward to this change mainly because I understood that it would allow us to have helm values files in our repository and then reference an external chart to use them with. 

This sounds like it will explicitly not be the case?

Reply:
Your good! Our docs will need to be clear on that. I hope 99% of folks will plan to host their external values file in git instead of Helm.